### ISSUE TITLE
Amplify deploys not happening for branches created from past commits

### ISSUE THREAD

**Issue by liberty-jacklin** [aws-amplify/amplify-hosting#3400]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d7kl3w7vdqcol

### AWS Region

eu-west-1

### Amplify Hosting feature

Deployments

### Describe the bug

We are finding that if we try and revert to a past commit and raise a PR for a branch with that commit (e.g. `git checkout <past-commit-hash> -b <branch-name>`), that it does not trigger an amplify deployment. 
It just continuously shows "Expected — Waiting for status to be reported" in Github and nothing shows up in the Amplify console. (Deploys are working perfectly fine for new branches with new commits)
<img width="854" alt="image" src="https://user-images.githubusercontent.com/118543165/229105468-4af24e31-139c-4472-a97d-126f75c20c6e.png">
I have looked at the webhooks but can't find anything, so any more information would be greatly appreciated :)
If it helps here is our setup:
 ```const amplifyApp = new amplify.App(this, 'Project Name Amplify', {
            appName: 'projectName',
            role: amplifyAppRole,
            autoBranchDeletion: true,
            autoBranchCreation: {
                patterns: ['*', '*/**'],
                pullRequestPreview: true
            },
            sourceCodeProvider: new amplify.GitHubSourceCodeProvider({
                owner,
                repository,
                oauthToken: SecretValue.unsafePlainText(props.githubToken)
            })
        })
```

   

### Expected behavior

I would expect to be able to raise a PR for a branch from a past commit and it would deploy as any other scenario does for us!

### Reproduction steps

1. from any branch, do `git checkout <commit-hash> -b <branch-name>`
2. push and raise a PR for that branch

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by hloriii** [comment]
Hi @liberty-jacklin 👋🏽 thanks for raising this issue. We are working on reproducing this behavior and determining next steps. If these commits have been previously deployed, try navigating to the build where the commit was used and hit the `Redeploy` button. We will update this issue again shortly.

**Comment by liberty-jacklin** [comment]
> Hi @liberty-jacklin 👋🏽 thanks for raising this issue. We are working on reproducing this behavior and determining next steps. If these commits have been previously deployed, try navigating to the build where the commit was used and hit the `Redeploy` button. We will update this issue again shortly.

Hey @hloriii thanks for getting back to me! So we now think it might be because of hitting our limit of branches (for some reason they haven't always been deleting on closing the PR), so we've tried deleting some and things work at the moment, not 100% sure if that was it but all good for now and will let you know within the next week if the problem arises again :)

**Comment by hloriii** [comment]
@liberty-jacklin Glad to hear! I was not able to reproduce the behavior on my end so that may have been the cause. I'll close this out since you have found the potential root of the problem and are unblocked. Please open a new issue if this behavior occurs again and we will be happy to investigate. Thank you!
